keanu-python/keanu/sequence.py

```diff
@@ -59,17 +64,30 @@ def __init__(self,
                 "Cannot create a sequence of an unknown size: you must specify either a count of a data_generator")
         elif count is not None and data_generator is not None:
             raise ValueError("If you pass in a data_generator you cannot also pass in a count")
+        elif factories is None:
+            raise ValueError("You must provide a value for the 'factories' input")
+
+        if not isinstance(factories, CollectionsIterable):
+            factories = [factories]
 
         if count is not None:
-            function = lambda p: factory(SequenceItem(p))
-            consumer = Consumer(function)
-            builder = builder.count(count).withFactory(consumer)
+
+            def wrap_function_argument_in_sequence_item(f, p):
+                return f(SequenceItem(p))
+
+            functions = [Consumer(partial(wrap_function_argument_in_sequence_item, f)) for f in factories]
+            java_functions = ListConverter().convert(functions, k._gateway._gateway_client)
+            builder = builder.count(count).withFactories(java_functions)
 
         if data_generator is not None:
-            bifunction = lambda p, data: factory(SequenceItem(p), data)
-            biconsumer = BiConsumer(bifunction)
+
+            def wrap_function_arguments_in_sequence_item(f, p, data):
+                return f(SequenceItem(p), data)
+
+            bifunctions = [BiConsumer(partial(wrap_function_arguments_in_sequence_item, f)) for f in factories]
```

Comment:
you can use a lambda here - no need for `wrap_function_argument_in_sequence_item`